### PR TITLE
download jenkins-cli.jar instead of extracting it from jenkins.war

### PR DESCRIPTION
**Pull request by hoomaan-kh**
… the Jenkins-cli.

https://github.com/voxpupuli/puppet-jenkins/issues/998

### CODE REVIEW

**Comment by ekohl** [comment]
This has a merge conflict. Could you make sure it cleanly applies?

**Comment by hoomaan-kh** [comment]
The conflict had been fixed

File: manifests/cli.pp [hunk 1]
```diff
@@ -36,7 +32,7 @@
     creates => $jar,
   }
   ~> exec { 'jenkins-cli' :
-    command     => "${extract_jar} && ${move_jar} && ${remove_dir}",
+    command     => "sleep 30 && ${extract_jar}",
```
Comment (kenyon) [inline manifests/cli.pp:35]: Why `sleep 30`?

File: manifests/cli.pp [hunk 2]
```diff
@@ -23,11 +23,7 @@
   }
 
   $jar = "${jenkins::libdir}/jenkins-cli.jar"
-  $extract_jar = "jar -xf ${jenkins::libdir}/jenkins.war WEB-INF/lib/"
-  $move_jar = "mv WEB-INF/lib/cli-*.jar ${jar}"
-  $remove_dir = 'rm -rf WEB-INF'
-  $cli_tries = $jenkins::cli_tries
-  $cli_try_sleep = $jenkins::cli_try_sleep
+  $extract_jar = "/bin/wget -O ${jar} http://127.0.0.1:8080/jnlpJars/jenkins-cli.jar"
```
Comment (kenyon) [inline manifests/cli.pp:26]: If we're not extracting anymore, we shouldn't call the variable `$extract_jar`.

Comment (kenyon) [inline manifests/cli.pp:26]: Should be `wget`, not `/bin/wget`.

**Review by kenyon — CHANGES_REQUESTED** [review]
I've rebased to remove the merge commit, and fixed the commit message, but I'm not sure about the actual changes.

File: manifests/cli.pp (hunk 2 above)
Comment (kenyon) [inline manifests/cli.pp:26]: The host part of the URL should be `localhost`, not `127.0.0.1`, so that we are not dependent on legacy IPv4 being available.

**Comment by kenyon** [comment]
For reference, the docs for this are here: https://www.jenkins.io/doc/book/managing/cli/#downloading-the-client

File: manifests/cli.pp (hunk 1 above)
Comment (hoomaan-kh) [reply manifests/cli.pp:35]: As it is downloading the "jenkins-cli" right after the Jenkins service is activated, we need to wait for a couple of seconds so the path will be accessible.

File: manifests/cli.pp (hunk 2 above)
Comment (hoomaan-kh) [reply manifests/cli.pp:26]: I'll change the name

**Comment by hoomaan-kh** [comment]
Do we need more changes or it could be merged?

**Comment by kenyon** [comment]
> Do we need more changes or it could be merged?

We need the tests to pass. I rebased on latest master and squashed the commits.

**Comment by ekohl** [comment]
The current solution does work and there hasn't been activity for a year. Closing, but feel free to open a new PR.

**Pull request closed without merging**